app/client/lib/ethereum/observeBlocks.js

```diff
@@ -19,49 +19,64 @@ Update wallet balances
 @method updateBalances
 */
 updateBalances = function() {
+    // UPDATE TOKEN BALANCES
+    var allWatchedAccounts = EthAccounts.find().fetch().concat(Wallets.find().fetch().concat(CustomContracts.find().fetch()));
 
-    var walletsAndContracts = Wallets.find().fetch().concat(CustomContracts.find().fetch());
+    // go through all existing accounts, for each token
+    _.each(allWatchedAccounts, function(account){
 
-    // UPDATE WALLETS ACCOUNTS balance
-    _.each(walletsAndContracts, function(wallet){
-        if(wallet.address) {
-            web3.eth.getBalance(wallet.address, function(err, res){
+        if(account.address) {
+            web3.eth.getBalance(account.address, function(err, res){
                 if(!err) {
                     // is of type wallet
-                    if(wallet.creationBlock) {
-                        Wallets.update(wallet._id, {$set: {
+                    if(account.creationBlock) {
+                        Wallets.update(account._id, {$set: {
                             balance: res.toString(10)
                         }});
                     } else {
-                        CustomContracts.update(wallet._id, {$set: {
+                        EthAccounts.update(account._id, {$set: {
+                            balance: res.toString(10)
+                        }});
+                        CustomContracts.update(account._id, {$set: {
                             balance: res.toString(10)
                         }});
                     }
                 }
             });
 
             // update dailylimit spent, etc, if wallet type
-            if(wallet.creationBlock) {
+            if(account.creationBlock) {
                 Meteor.setTimeout(function() {
-                    updateContractData(wallet);
+                    updateContractData(account);
                 }, 1000);
             }
         }
-    });
 
 
-    
-    // UPDATE TOKEN BALANCES
-    var walletsContractsAndAccounts = EthAccounts.find().fetch().concat(walletsAndContracts);
+        // Only check ENS names every N minutes
+        if (!account.ensCheck || (account.ensCheck && Date.now() - account.ensCheck > 10*60*1000)) {
+            Helpers.getENSName(account.address, function(err, name, returnedAddr) {
+                if (!err && account.address.toLowerCase() == returnedAddr){
+                    EthAccounts.update({address: account.address}, {$set:{ name: name, ens: true, ensCheck: Date.now()}});
```

Comment:
I guess its more performer to get the date once, safe it in a variable and then use it in all the below places.